pandas/core/arrays/_mixins.py

```diff
@@ -176,9 +182,14 @@ def _concat_same_type(
         return to_concat[0]._from_backing_data(new_values)  # type: ignore[arg-type]
 
     @doc(ExtensionArray.searchsorted)
-    def searchsorted(self, value, side="left", sorter=None):
-        value = self._validate_searchsorted_value(value)
-        return self._ndarray.searchsorted(value, side=side, sorter=sorter)
+    def searchsorted(
+        self,
+        value: ArrayLike | object,
```

Comment:
object -> Scalar?

Reply:
Would violate the Liskov substitution principle.  Our `searchsorted` needs to have the EA's reject an incompatible type

Reply:
I guess that would be bc you've typed EA.searchsorted with object, so why not change it there?

Reply:
Because in `EA`, we have to allow any type to be inside the `EA`, including non-EA types.